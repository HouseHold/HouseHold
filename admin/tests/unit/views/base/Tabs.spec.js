import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue'
import Tabs from '@/views/base/Tabs'

Vue.use(CoreuiVue)

describe('Tabs.vue', () => {
  it('has a name', () => {
    expect(Tabs.name).toBe('Tabs')
  })
  it('has a created hook', () => {
    expect(typeof Tabs.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Tabs.data).toMatch('function')
  })
  it('is Vue instance', () => {
    const wrapper = mount(Tabs)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Tabs', () => {
    const wrapper = mount(Tabs)
    expect(wrapper.is(Tabs)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Tabs)
    expect(wrapper.element).toMatchSnapshot()
  })
})
