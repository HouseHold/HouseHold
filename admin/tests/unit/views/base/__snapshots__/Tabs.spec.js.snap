// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <ccard-stub>
        <ccardheader-stub>
          
          Tabs
          
          <div
            class="card-header-actions"
          >
            <a
              class="card-header-action"
              href="https://coreui.io/vue/docs/components/tabs"
              rel="noreferrer noopener"
              target="_blank"
            >
              <small
                class="text-muted"
              >
                docs
              </small>
            </a>
          </div>
        </ccardheader-stub>
         
        <ccardbody-stub>
          <ctabs-stub
            fade="true"
            variant="tabs"
          >
            <ctab-stub
              active="true"
              title="Home"
            >
              
              1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub
              title="Profile"
            >
              
              2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub
              disabled="true"
              title="Disabled"
            >
              
              3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
          </ctabs-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <ccard-stub>
        <ccardheader-stub>
          
          Tabs
        
        </ccardheader-stub>
         
        <ccardbody-stub>
          <ctabs-stub
            fade="true"
            variant="pills"
          >
            <ctab-stub
              active="true"
              title="Home"
            >
              
              1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub
              title="Profile"
            >
              
              2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub
              disabled="true"
              title="Disabled"
            >
              
              3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
          </ctabs-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <ccard-stub>
        <ccardheader-stub>
          
          Tabs with icons
        
        </ccardheader-stub>
         
        <ccardbody-stub>
          <ctabs-stub
            fade="true"
            variant="tabs"
          >
            <ctab-stub
              active="true"
            >
              <template>
                <cicon-stub
                  name="cil-calculator"
                />
              </template>
              
              1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub>
              <template>
                <cicon-stub
                  name="cil-basket"
                />
              </template>
              
              2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub>
              <template>
                <cicon-stub
                  name="cil-chart-pie"
                />
              </template>
              
              3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
          </ctabs-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <ccard-stub>
        <ccardheader-stub>
          
          Tabs with icons
        
        </ccardheader-stub>
         
        <ccardbody-stub>
          <ctabs-stub
            addtabclasses="mt-1"
            fade="true"
            variant="tabs"
          >
            <ctab-stub
              active="true"
            >
              <template>
                <cicon-stub
                  name="cil-calculator"
                />
                 Calculator
              
              </template>
              
              1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub>
              <template>
                <cicon-stub
                  name="cil-basket"
                />
                 Shopping cart
              
              </template>
              
              2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub>
              <template>
                <cicon-stub
                  name="cil-chart-pie"
                />
                 Charts
              
              </template>
              
              3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
          </ctabs-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      tag="div"
      xs="12"
    >
      <ccard-stub>
        <ccardheader-stub>
          
          Tabs vertical
        
        </ccardheader-stub>
         
        <ccardbody-stub>
          <ctabs-stub
            fade="true"
            variant="pills"
            vertical="true"
          >
            <ctab-stub
              active="true"
            >
              <template>
                <cicon-stub
                  name="cil-calculator"
                />
                 Calculator
              
              </template>
              
              1. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub>
              <template>
                <cicon-stub
                  name="cil-basket"
                />
                 Shopping cart
              
              </template>
              
              2. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
             
            <ctab-stub>
              <template>
                <cicon-stub
                  name="cil-chart-pie"
                />
                 Charts
              
              </template>
              
              3. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore
              et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
              dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
              officia deserunt mollit anim id est laborum.
            
            </ctab-stub>
          </ctabs-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
  </crow-stub>
</div>
`;
